### PR TITLE
admin settings pagination 

### PR DESCRIPTION
**Pull request by JanAckermann**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/9048

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Follow-ups:
- Merge both `usePagination` composables into one and move it to `web-pkg`
- Reset pagination (or `currentPage - 1`) when all items on the last page have been deleted
- Use constants for the events

### CODE REVIEW

File: packages/web-app-admin-settings/src/components/Groups/GroupsList.vue [hunk 1]
```diff
@@ -28,7 +28,11 @@
           :label="$gettext('Select all groups')"
           :model-value="allGroupsSelected"
           hide-label
-          @update:model-value="$emit('toggleSelectAllGroups')"
+          @update:model-value="
+            allGroupsSelected
+              ? $emit('unSelectAllGroups')
```
Comment (lookacat) [inline packages/web-app-admin-settings/src/components/Groups/GroupsList.vue]: is it possible to extract the event into a const?

Comment (JanAckermann) [reply packages/web-app-admin-settings/src/components/Groups/GroupsList.vue]: why?

File: packages/web-app-admin-settings/src/components/Groups/GroupsList.vue [hunk 2]
```diff
@@ -183,6 +200,36 @@ export default defineComponent({
...
+        b = b[prop]?.toString() || ''
+        return desc ? b.localeCompare(a) : a.localeCompare(b)
+      })
+    }
+
+    const items = computed(() => {
+      return orderBy(
+        filter(props.groups, unref(filterTerm)),
+        unref(sortBy),
+        unref(sortDir) === 'desc'
```
Comment (kulmann) [inline packages/web-app-admin-settings/src/components/Groups/GroupsList.vue:223]:
```suggestion
        unref(sortDir) === SortDir.Desc
```

File: packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue [hunk 3]
```diff
@@ -206,9 +213,20 @@ export default defineComponent({
           : a.localeCompare(b, undefined, { numeric })
       })
     }
-    const orderedSpaces = computed(() =>
+    const items = computed(() =>
       orderBy(filter(props.spaces, unref(filterTerm)), unref(sortBy), unref(sortDir) === 'desc')
```
Comment (kulmann) [inline packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue:217]:
```suggestion
      orderBy(filter(props.spaces, unref(filterTerm)), unref(sortBy), unref(sortDir) === SortDir.Desc)
```

File: packages/web-app-admin-settings/src/components/Users/UsersList.vue [hunk 4]
```diff
@@ -134,13 +141,20 @@ export default defineComponent({
...
   },
-  emits: ['unSelectAllUsers', 'toggleSelectAllUsers', 'toggleSelectUser'],
+  emits: ['unSelectAllUsers', 'selectUsers', 'toggleSelectUser'],
   setup(props, { emit }) {
+    const { $gettext } = useGettext()
+
     const contextMenuButtonRef = ref(undefined)
+    const filterTerm = ref<string>('')
+    const sortBy = ref<string>('onPremisesSamAccountName')
+    const sortDir = ref<string>('asc')
```
Comment (kulmann) [inline packages/web-app-admin-settings/src/components/Users/UsersList.vue:151]:
```suggestion
    const sortDir = ref<string>(SortDir.Asc)
```

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-admin-settings/src/components/Groups/GroupsList.vue (hunk 1 above)
Comment (kulmann) [reply packages/web-app-admin-settings/src/components/Groups/GroupsList.vue]: same reason why strings are often put into constants: to make it harder to mess up typing the string correctly. :trollface: referencing a variable is always easier than looking up the correct string...

File: packages/web-app-admin-settings/src/composables/usePagination.ts
[the remark this answers is not in the document]
Comment (JanAckermann) [reply packages/web-app-admin-settings/src/composables/usePagination.ts:1]: Hmmm ours is technically a bit different and also does the client side pagination

File: packages/web-app-admin-settings/src/components/Groups/GroupsList.vue (hunk 1 above)
Comment (JanAckermann) [reply packages/web-app-admin-settings/src/components/Groups/GroupsList.vue]: I see... we use them multiple time. We can make this happen in this pr, even if this is not a refactoring pr 😅

So we also wanted to keep it as small as possible for you guys to review.

But you get what you asked for 😜🖤

File: packages/web-app-admin-settings/src/composables/usePagination.ts
[the remark this answers is not in the document]
Comment (JammingBen) [reply packages/web-app-admin-settings/src/composables/usePagination.ts:1]: As discussed via chat: I will follow-up on this in a separate PR to master.

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/36162/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36162/tracing/user-group-assignments-can-be-handled-via-batch-actions-admin-2023-5-30-09-58-35.zip 


</p></details>
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

**Review by kulmann — APPROVED** [review]

**Review by lookacat — APPROVED** [review]
Code LGTM, works like intended 👍🏼

**Pull request merged**